zebra-chain/src/orchard/tree.rs

```diff
@@ -317,6 +339,46 @@ impl NoteCommitmentTree {
         }
     }
 
+    /// Returns true if the most recently appended leaf completes the subtree
+    pub fn is_complete_subtree(&self) -> bool {
+        self.inner.value().map_or(false, |x| {
+            x.position()
+                .is_complete_subtree(TRACKED_SUBTREE_HEIGHT.into())
+        })
+    }
+
+    /// Returns subtree index at [`TRACKED_SUBTREE_HEIGHT`]
+    pub fn subtree_index(&self) -> u16 {
+        (self.position() >> TRACKED_SUBTREE_HEIGHT)
+            .try_into()
+            .expect("always within u16")
+    }
```

Comment:
I strongly recommend using the same APIs as `zcashd`, rather than creating our own duplicate implementations.

Since we have a subtree index wrapper type, it might help to use it here as well.

Untested code:

```suggestion
    pub fn subtree_index(&self) -> NoteCommitmentSubtreeIndex
        incrementalmerkletree::Address::above_position(
                TRACKED_SUBTREE_HEIGHT,
                self.position()
            )
            .index()
            .try_into()
            .expect("always within u16")
```

Reply:
This seems excessively verbose:

```suggestion
    pub fn subtree_index(&self) -> u16 {
        incrementalmerkletree::Address::above_position(
            TRACKED_SUBTREE_HEIGHT.into(),
            self.inner.value().map_or(0.into(), |x| x.position()),
        )
        .index()
        .try_into()
        .expect("always within u16")
    }
```